### ISSUE TITLE
dist_hypertable test is flaky

### ISSUE THREAD

**Issue by svenklemm** [timescale/timescaledb#3090]
dist_hypertable fails alot across all postgres versions with the recently add test for chunks_in and ChunkAppend

Example: https://github.com/timescale/timescaledb/pull/3087/checks?check_run_id=2270557654

Test diff:
```
diff -u /__w/timescaledb/timescaledb/tsl/test/expected/dist_hypertable-12.out /__w/timescaledb/timescaledb/build/tsl/test/results/dist_hypertable-12.out
--- /__w/timescaledb/timescaledb/tsl/test/expected/dist_hypertable-12.out	2021-04-05 14:57:29.747509253 +0000
+++ /__w/timescaledb/timescaledb/build/tsl/test/results/dist_hypertable-12.out	2021-04-05 15:03:28.346082692 +0000
@@ -718,18 +718,17 @@
                              Remote SQL: SELECT "time" FROM public.disttable WHERE _timescaledb_internal.chunks_in(public.disttable.*, ARRAY[2, 1]) AND (("time" IS NOT NULL)) ORDER BY "time" DESC NULLS FIRST LIMIT 1
                              Remote EXPLAIN: 
                                Limit
-                                 Output: disttable."time"
-                                 ->  Custom Scan (ChunkAppend) on public.disttable
-                                       Output: disttable."time"
-                                       Order: disttable."time" DESC
-                                       Startup Exclusion: false
-                                       Runtime Exclusion: false
-                                       ->  Index Only Scan using _dist_hyper_1_5_chunk_disttable_time_idx on _timescaledb_internal._dist_hyper_1_5_chunk
-                                             Output: _dist_hyper_1_5_chunk."time"
-                                             Index Cond: (_dist_hyper_1_5_chunk."time" IS NOT NULL)
-                                       ->  Index Only Scan using _dist_hyper_1_3_chunk_disttable_time_idx on _timescaledb_internal._dist_hyper_1_3_chunk
-                                             Output: _dist_hyper_1_3_chunk."time"
-                                             Index Cond: (_dist_hyper_1_3_chunk."time" IS NOT NULL)
+                                 Output: _dist_hyper_1_5_chunk."time"
+                                 ->  Sort
+                                       Output: _dist_hyper_1_5_chunk."time"
+                                       Sort Key: _dist_hyper_1_5_chunk."time" DESC
+                                       ->  Append
+                                             ->  Seq Scan on _timescaledb_internal._dist_hyper_1_5_chunk
+                                                   Output: _dist_hyper_1_5_chunk."time"
+                                                   Filter: (_dist_hyper_1_5_chunk."time" IS NOT NULL)
+                                             ->  Seq Scan on _timescaledb_internal._dist_hyper_1_3_chunk
+                                                   Output: _dist_hyper_1_3_chunk."time"
+                                                   Filter: (_dist_hyper_1_3_chunk."time" IS NOT NULL)
  
                        ->  Custom Scan (DataNodeScan) on public.disttable disttable_3
                              Output: disttable_3."time"
@@ -751,7 +750,7 @@
                                              Output: _dist_hyper_1_2_chunk."time"
                                              Index Cond: (_dist_hyper_1_2_chunk."time" IS NOT NULL)
  
-(69 rows)
+(68 rows)
 
 EXPLAIN (VERBOSE, COSTS FALSE)
 SELECT max(temp)
```

**Comment by k-rus** [comment]
I've got different error in [this build](https://github.com/timescale/timescaledb/pull/3091/checks?check_run_id=2276190169)
```diff
+ERROR:  [db_dist_hypertable_1]: relation "disttable" does not exist
```

**Comment by k-rus** [comment]
The test is extremely flaky and fails almost every build.

**Comment by nikkhils** [comment]
This query 

`SELECT max(time)                                                                
FROM disttable;`

is producing the flakiness. Looks like some datanodes do not have enough data to enforce the index scan.

**Comment by nikkhils** [comment]
And now it's failing for completely weird reasons on CI. I am trying to reproduce failure by running the test in a loop locally and on an Ubuntu VM, but no luck so far. Wonder why it's so flaky in CI..

**Comment by nikkhils** [comment]
Fixed via #3152
